File: backend/product_service/src/main/java/ucv/app_inventory/domain/entities/Category.java
package ucv.app_inventory.domain.entities;

import java.io.Serializable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "Categories")
public class Category implements Serializable {

    @NotNull
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = true)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "ENUM('ACTIVE', 'INACTIVE') DEFAULT 'ACTIVE'")
    private Status status;

    public enum Status {
        ACTIVE,
        INACTIVE
    }
    @PrePersist
    public void prePersist() {
        if (status == null) {
            status = Category.Status.ACTIVE;
        }
    }
}

File: frontend/src/products/domain/models/Category.js
class Category {
    constructor({ id, name, description, status }) {
      this.id = id || null;
      this.name = name || '';
      this.description = description || '';
      this.status = status || 'ACTIVE'; // ACTIVE, INACTIVE
    }
  
    activate() {
      this.status = 'ACTIVE';
    }
  
    deactivate() {
      this.status = 'INACTIVE';
    }
  }
  
  export default Category;
  